### DISCUSSION TITLE
AnalyticalTable - change data

### DISCUSSION THREAD

**Discussion by Qcyk** [UI5/webcomponents-react#5487] [Q&A] [answered]
Hi, 

Is it possible to change a data (add new row) and remember which rows were selected using Analytical Table and selectedRows Is? Always when we changes dynamicaly the data all rows are deselecting.

**Comment by Lukas742** [comment, marked as answer]
Hi @Qcyk 

please see this recipe: https://sap.github.io/ui5-webcomponents-react/?path=/docs/data-display-analyticaltable-recipes--docs#how-to-stop-the-table-state-from-automatically-resetting-when-the-data-changes

**Reply by Qcyk** [reply]
Thank you, 

I have already added two useMemo functions. 
1. Update data of Analytical Table which return new array
2. Update SelectionRowIds after data changed. 

![image](https://github.com/SAP/ui5-webcomponents-react/assets/25754184/2585bbe4-785c-40af-b641-bfe22c17d0cf)

It also works.
